docs/end_to_end_tutorials/principled_dev_practices.md

```diff
@@ -0,0 +1,94 @@
+# Principled Development Practices
+
+In order to develop your application, it can help to implement some principled development practices.
+
+We've boiled down the key practices into 3 pillars.
+
+We will also try to do a compare and contrast with traditional software engineering and ML development, highlighting similarities and differences.
+
+Here are the main pillars of principled development of LLM and RAG applications:
+
+```{toctree}
+---
+maxdepth: 1
+---
+/end_to_end_tutorials/dev_practices/observability.md
+/end_to_end_tutorials/dev_practices/evaluation.md
+/end_to_end_tutorials/dev_practices/monitoring.md
+/end_to_end_tutorials/dev_practices/UX_patterns.md
```

Comment:
nit: is this file checked in? I don't see it

Reply:
It's not but I thought I'd just leave it in as a TODO. It seemed kind of relevant as the UX seems pretty important for LLM apps.

Reply:
can we not include the link then? it'll just trigger doc errors and not actually do anything user facing. can always add once we have the doc (if you want to leave a comment in markdown you can do that too, just not add an actual broken link)